auth/user_mgt.go

```diff
@@ -0,0 +1,553 @@
+	it := &UserIterator{
+		ctx:    ctx,
+		client: c,
+	}
+	it.pageInfo, it.nextFunc = iterator.NewPageInfo(
+		it.fetch,
+		func() int { return len(it.users) },
+		func() interface{} { b := it.users; it.users = nil; return b })
+	it.pageInfo.MaxSize = maxReturnedResults
+	it.pageInfo.Token = startToken
+	return it
+}
+
+func (it *UserIterator) fetch(pageSize int, pageToken string) (string, error) {
+	params := map[string]interface{}{"maxResults": pageSize}
+	if pageToken != "" {
+		params["nextPageToken"] = pageToken
+	}
+
+	var lur listUsersResponse
+	err := it.client.makeHTTPCall(it.ctx, "downloadAccount", params, &lur)
+	if err != nil {
+		return "", err
+	}
+	for _, u := range lur.Users {
+		eu, err := makeExportedUser(u)
+		if err != nil {
+			return "", err
+		}
+		it.users = append(it.users, eu)
+	}
+	it.pageInfo.Token = lur.NextPage
+	return lur.NextPage, nil
+}
+
+// PageInfo supports pagination. See the google.golang.org/api/iterator package for details.
+// Page size can be determined by the NewPager(...) function described there.
+func (it *UserIterator) PageInfo() *iterator.PageInfo { return it.pageInfo }
+
+// Next returns the next result. Its second return value is [iterator.Done] if
+// there are no more results. Once Next returns [iterator.Done], all subsequent
+// calls will return [iterator.Done].
+func (it *UserIterator) Next() (*ExportedUserRecord, error) {
+	if err := it.nextFunc(); err != nil {
+		return nil, err
+	}
+	user := it.users[0]
+	it.users = it.users[1:]
+	return user, nil
+}
+
+// SetCustomUserClaims sets additional claims on an existing user account.
+//
+// Custom claims set via this function can be used to define user roles and privilege levels.
+// These claims propagate to all the devices where the user is already signed in (after token
+// expiration or when token refresh is forced), and next time the user signs in. The claims
+// can be accessed via the user's ID token JWT. If a reserved OIDC claim is specified (sub, iat,
+// iss, etc), an error is thrown. Claims payload must also not be larger then 1000 characters
+// when serialized into a JSON string.
+func (c *Client) SetCustomUserClaims(ctx context.Context, uid string, customClaims map[string]interface{}) error {
+	if customClaims == nil || len(customClaims) == 0 {
+		customClaims = map[string]interface{}{}
+	}
+	_, err := c.UpdateUser(ctx, uid, (&UserToUpdate{}).CustomClaims(customClaims))
+	return err
+}
+
+// ------------------------------------------------------------
+// Utilities for Create and Update input structs.
+
+func (p *commonParams) set(key string, value interface{}) {
+	if p.payload == nil {
+		p.payload = make(map[string]interface{})
+	}
+	p.payload[key] = value
+}
+
+// assumes that payloadName is a string field in p.payload
+func processDeletion(p *commonParams, payloadName string) error {
+	if dn, ok := p.payload[payloadName]; ok && len(dn.(string)) == 0 {
```

Comment:
It looks like this function only ever returns nil. Remove the return value, if that's the case.

Also what your criteria for making a function a member of `commonParams` type, as opposed to a regular top-level function. Make sure you follow some consistent semantics.